service/storage_proxy.cc

```diff
@@ -5385,9 +5385,11 @@ class speculating_read_executor : public abstract_read_executor {
                 auto send_request = [&] (bool has_data) {
                     if (has_data) {
                         _proxy->get_stats().speculative_digest_reads++;
+                        tracing::trace(_trace_state, "Launching speculative retry for digest");
```

Comment:
Can we include the target endpoint here (and below)? I know one should be able to deduce from the target list, but more explicit is more good.

Reply:
Actually nevermind, there is a "Added extra target 127.0.0.1 for speculative read" a few lines above.